### PR TITLE
Treatment options criteria

### PR DESCRIPTION
**Pull request by gregquinn2001**
For JIRA 1457. Updated presentation of Julia's generic filtering support for sections with Targeted Data Panel. Uses chips to show active criteria with an x to remove them and then can expand to see checkboxes for all possible criteria. Timeline over the counter meds now uses it as does timeline now. No slider implemented - only the advanced checkbox method to start. Changing criteria causes a refresh but data doesn't actually change since we are in demo mode. criteria will need to be a method because Treatment Options will need to call service so it can figure our criteria for condition based on available data.

_Pull requests into Flux require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:._

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome
- [ ] Manually tested in IE

**Documentation**

- [x] This pull request describes why these changes were made
- [ ] Recognizes any potential shortcomings/bugs in the description 
- [ ] Demo script is updated 
- [ ] Documentation on Wiki has been updated 
- [ ] Additional technical components and libraries are documented and added to wiki as needed

**Code Quality**

- [X] 4-space indents - convert any tabs to spaces
- [X] Use public class field syntax for binding functions - ex. handleChange = () => { ... };
- [X] Code is commented

**Tests**

- [X] Existing tests passed

## Reviewer 1: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

## Reviewer 2: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by sgober** [comment]
As I remove treatment options from the list the graph reloads but none of the data changes.  Is that how this is supposed to function?

**Comment by gregquinn2001** [comment]
Yes, the graph is a demo right now and the filtering has no effect. The critiera would be passed to the service when we switch to the real thing.

**Comment by gregquinn2001** [comment]
rebased including merging @karenzheung changes to save section filters in preference manager. Saving and loading of filters is now handled in TargetDataSection and doesn't require any support in metadata. Data retrieval methods can retrieve filter values via a provided getFilterValue function.

File: src/summary/metadata/TreatmentOptionsSection.jsx [hunk 1]
```diff
@@ -21,17 +21,71 @@ export default class TreatmentOptionsSection extends MetadataSection {
...
+                                propertyValueFunction: (patient, condition) => { return condition.hasPastTreatment('C0851238', patient) } }
+                    ],
                     itemsFunction: this.getTreatmentData
                 }
             ]
         };
     }
 
-    getTreatmentData = (patient, condition, subsection) => {
+    // getTreatmentCriteria = (patient, condition) => {
```
Comment (danlee1025) [inline src/summary/metadata/TreatmentOptionsSection.jsx]: A lot of commented out code in this file(lines 51-62 and lines 69 - 88.  Do we want to keep these comments in here for later use?

Other than that, the criteria options look great!

Comment (gregquinn2001) [reply src/summary/metadata/TreatmentOptionsSection.jsx]: I was thinking to keep that code as we will need to ask the service for criteria that make sense given a condition.

Comment (danlee1025) [reply src/summary/metadata/TreatmentOptionsSection.jsx]: ok cool I will approve and merge then!

**Pull request merged**